ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -921,33 +955,38 @@ private void sendGetResponse(RequestResponseChannel requestResponseChannel, GetR
    * Check that the provided partition is valid, on the disk, and can be written to.
    * @param partition the partition to validate.
    * @param requestType the {@link RequestOrResponseType} being validated.
+   * @param skipPartitionAndDiskAvailableCheck whether to skip ({@code true}) conditions check for the availability of
+   *                                           partition and disk.
    * @return {@link ServerErrorCode#No_Error} error if the partition can be written to, or the corresponding error code
    *         if it cannot.
    */
-  private ServerErrorCode validateRequest(PartitionId partition, RequestOrResponseType requestType) {
+  private ServerErrorCode validateRequest(PartitionId partition, RequestOrResponseType requestType,
+      boolean skipPartitionAndDiskAvailableCheck) {
     // 1. Check partition is null
     if (partition == null) {
       metrics.badRequestError.inc();
       return ServerErrorCode.Bad_Request;
     }
-    // 2. check if partition exists on this node and that the store for this partition has been started
-    if (storageManager.getStore(partition) == null) {
-      if (partitionsInCurrentNode.contains(partition)) {
-        metrics.diskUnavailableError.inc();
-        return ServerErrorCode.Disk_Unavailable;
-      } else {
-        metrics.partitionUnknownError.inc();
-        return ServerErrorCode.Partition_Unknown;
-      }
-    }
-    // 3. ensure the disk for the partition/replica is available
-    List<? extends ReplicaId> replicaIds = partition.getReplicaIds();
-    for (ReplicaId replica : replicaIds) {
-      if (replica.getDataNodeId().getHostname().equals(currentNode.getHostname())
-          && replica.getDataNodeId().getPort() == currentNode.getPort()) {
-        if (replica.getDiskId().getState() == HardwareState.UNAVAILABLE) {
+    if (!skipPartitionAndDiskAvailableCheck) {
+      // 2. check if partition exists on this node and that the store for this partition has been started
+      if (storageManager.getStore(partition) == null) {
+        if (partitionsInCurrentNode.contains(partition)) {
           metrics.diskUnavailableError.inc();
```

Comment:
As discussed, Disk_Unavailable is not appropriate here. Perhaps this should be `Partition_Unavailable` (so that a caller does not think that all partitions on the said disk are down).

Reply:
Will introduce `Partition_Unavailable` error code in a new PR.